pages/box_buster.vue

```diff
@@ -0,0 +1,171 @@
+<template>
+  <b-container>
+    <h1>Box Buster</h1>
+    <p class="lead">
+      Quickly overview box contents to assist with cherrypicking
+    </p>
+    <b-form-group
+      id="box-barcode"
+      description="Scan in a lighthouse box barcode to show information about all plates contained in the box."
+      label="Please scan lighthouse box barcode"
+      label-for="box-barcode-field"
+      :invalid-feedback="labwhereFeedback"
+      valid-feedback="Box found"
+      :state="labwhereState"
+    >
+      <b-form-input
+        id="box-barcode-field"
+        v-model="barcode"
+        type="search"
+        trim
+        :state="labwhereState"
+        @change="findBox"
+      >
+      </b-form-input>
+    </b-form-group>
+    <b-table
+      v-if="showTable"
+      :items="plates"
+      :tbody-tr-class="rowClass"
+      show-empty
+      :empty-text="lighthouseFeedback"
+      sort-by="number_of_positives"
+      :sort-desc="true"
+      :fields="fields"
+    >
+      <template #table-caption>
+        <span>Box Summary:</span>
+        <span>Total of {{ total }} plates in box;</span>
+        <span>{{ total_with_maps }} plates with plates maps,</span>
+        <span>{{ total_without_maps }} without.</span>
+        <span>Total {{ total_positives }} positives.</span>
+      </template>
+    </b-table>
+  </b-container>
+</template>
+
+<script>
+import lighthouse from '../modules/lighthouse_service'
+import { getPlatesFromBoxBarcodes } from '@/modules/labwhere'
+
+// import lighthouse from '../modules/lighthouse_service'
+const countWithMap = (accumulator, plate) =>
+  accumulator + (plate.plate_map ? 1 : 0)
+const countWithoutMap = (accumulator, plate) =>
+  accumulator + (plate.plate_map ? 0 : 1)
+const sumPositives = (accumulator, plate) =>
+  accumulator + plate.number_of_positives
+const mapFormatter = (value) => (value ? 'Yes' : 'No')
+const countFormatter = (value, _key, item) => (item.plate_map ? value : 'N/A')
+const extractError = (response) => {
```

Comment:
I'm not really clear on what this does. Is there some edge case where an error is not returned? Did you see something in the tests that wasn't making sense?

Reply:
getPlatesFromBoxBarcodes returns both strings and Error objects as the error value of the attribute. Vue starts throwing up warnings if you pass an error object directly into a prop that is expecting a string. I can update the behaviour of getPlatesFromBoxBarcodes, but was away that was used in a fe other places.

Reply:
Some of the additional handling was just to avoid the risk of silent errors, but is probably over-cautious.

Reply:
Ah ok. If you look in index.vue we handle it slightly differently. So we have an alert message rather than an error message or success message. Then we have alerts each of which shows dependent on the status. Then if the error is an object it doesn't matter it just shows it.

Reply:
Yeah, just remembered another issue about using the form feedback parameters, is that it still validates them, even if they aren't used. (Hence the defaultResponse.error bit). So annoyingly we need to return valid content for the field, even if there is no error.